### DISCUSSION TITLE
Compatibility of KC versions with PostgreSQL versions

### DISCUSSION THREAD

**Discussion by semangard** [keycloak/keycloak#9232] [Q&A]
Hello,

Is there any documentation indicating wich the compliance between KeyCloak versions and Postgre versions ?
For example is its possible to run:
→ KC 16 + Postgre13
→ KC 16 + Postgre 9
Thanks and regards

**Comment by myfirstnameispaul** [comment]
They now have it [in the documentation](https://www.keycloak.org/server/db):

Database | Tested Version
-- | --
mariadb | 10
mssql | 2016
mysql | 8
oracle | 12c
postgres | 10

I'm also wondering what happens when I run it with PostgreSQL 13.

**Reply by semangard** [reply]
https://github.com/keycloak/keycloak/issues/10767

**Reply by dasniko** [reply]
> I'm also wondering what happens when I run it with PostgreSQL 13.

Nothing special, it just ... works!

**Reply by myfirstnameispaul** [reply]
@dasniko I feel like I'm missing something somewhere.

Based on the [link](https://jdbc.postgresql.org/documentation/changelog.html#version_all) in the issue, I was concerned that the JDBC driver (42.2.5) would preclude using anything later than 10, as support for even 11 wasn't added until 42.2.6.

**Reply by dasniko** [reply]
Typically, a driver supporting a newer version, and thus the features of a newer version, does not prevent an older driver version to work with a newer db version. You just can't use the new features.

Additionally, even though the docs just mention PG 10, the driver shipped with KC 17 is 42.3.1

I've running a production environment with PG 13 without problems, also with older KC versions (since 14 or so). 🤷‍♂️
Just because KC is only tested with PG 10, it doesn't mean that it won't work with newer versions. I think, we can't expect the team to test KC against all possible database versions, that would be too much effort for an open source software you get for free!

**Comment by tdduydev** [comment]
I am using KC 21.1.2 fine with PG 15

**Reply by christian-2** [reply]
In my case it's currently Keycloak 21.1.1 with PostgreSQL 15.3.

**Comment by novatskij** [comment]
Hello!
Is there any chance that KC 21.x.x is tested with PG 14.x and versions are compatible?
Any evidences that KC 21.x.x does not use PG 15 only features are highly appreciated.

I have observed the documentation. Unfortunately, no any info related to my questions has been found
https://www.keycloak.org/server/db

**Comment by christian-2** [comment]
I'd be optimistic also regarding PostgreSQL 14, for Hibernate should be able to bridge across most differences between DB versions. On the other hand, the relevant [Keycloak guide](https://www.keycloak.org/server/db) specifically recommends PostgreSQL 15. A way to garner concrete evidence would perhaps be to run any version of PostgresSQL against the [relevant tests](https://github.com/keycloak/keycloak/blob/main/docs/tests-db.md).

**Comment by hwtcl** [comment]
How can Keycloak latest version 23 support PostgreSQL former versions 9.2 and 9.6? What should I do?

**Reply by hwtcl** [reply]
Because I must support legacy system but want to upgrade KC23

**Comment by prasannavemula** [comment]
Hi,
Does keycloak version 26 supports PostgreSQL version 16? or If not, can i get details on which keycloak version  is compatible with PostgreSQL 16?

**Reply by stianst** [reply]
https://www.keycloak.org/server/db

**Reply by jon-nfc** [reply]
> https://www.keycloak.org/server/db

is there a kc to postgres version matrix? this doc shows one version for postgres and no mention of kc version.

I'm curios of what kc versions support postgres 15 as v15 this a hard limit for us due to other dependencies.

**Reply by ZhdArtemiy** [reply]
@jon-nfc Hi, can you help me. Have you tested KeyCloak v26 with Postgres v15 or v16?

**Reply by jon-nfc** [reply]
> @jon-nfc Hi, can you help me. Have you tested KeyCloak v26 with Postgres v15 or v16?

not yet, still stuck on v13 due to other apps.

**Comment by kotnikd3** [comment]
Hi, when can we expect support for Postgres 17?

**Reply by kotnikd3** [reply]
Postgres 17 is now supported from Keycloak version 26.1.0.
